src/cogent3/evolve/likelihood_tree.py

```diff
@@ -177,10 +169,11 @@ class _PyLikelihoodTreeEdge(_LikelihoodTreeEdge):
     LOG_BASE = numpy.log(BASE)
 
     def sum_input_likelihoodsR(self, result, *likelihoods):
-        result[:] = 1.0
-        for (i, index) in enumerate(self.indexes):
-            result *= numpy.take(likelihoods[i], index, 0)
-        return result
+        return likelihood_tree_numba.sum_input_likelihoods(
+            numpy.ascontiguousarray(self.indexes),
```

Comment:
given `self.indexes` is an attribute, call `numpy.ascontiguousarray` on creation of that attribute. But you will have to leave the `result` conversion as is